File: Klanvision_backend/src/main/java/com/klanvision/backend/model/Application.java
package com.klanvision.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "job_applications")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Application {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String jobTitle;
    private String name;
    private String dob;
    private String email;
    private String phone;
    private String gender;
    private String qualification;
    private String experience;

    @Column(columnDefinition = "TEXT")
    private String skills;

    private String linkedin;
    private String portfolio;

    @Lob
    @Column(columnDefinition = "LONGBLOB")
    private byte[] resumeData;
    
    private String resumeFileName;
    private String resumeContentType;

    private LocalDateTime submittedAt;

    @PrePersist
    protected void onCreate() {
        submittedAt = LocalDateTime.now();
    }
}

File: Klanvision_backend_node/models/Application.js
import { DataTypes } from 'sequelize';
import sequelize from '../config/db.js';

const Application = sequelize.define('Application', {
  id: {
    type: DataTypes.BIGINT,
    primaryKey: true,
    autoIncrement: true
  },
  jobTitle: {
    type: DataTypes.STRING,
    field: 'job_title'
  },
  name: {
    type: DataTypes.STRING
  },
  dob: {
    type: DataTypes.STRING
  },
  email: {
    type: DataTypes.STRING
  },
  phone: {
    type: DataTypes.STRING
  },
  gender: {
    type: DataTypes.STRING
  },
  qualification: {
    type: DataTypes.STRING
  },
  experience: {
    type: DataTypes.STRING
  },
  skills: {
    type: DataTypes.TEXT
  },
  linkedin: {
    type: DataTypes.STRING
  },
  portfolio: {
    type: DataTypes.STRING
  },
  resumeData: {
    type: DataTypes.BLOB('long'),
    field: 'resume_data'
  },
  resumeFileName: {
    type: DataTypes.STRING,
    field: 'resume_file_name'
  },
  resumeContentType: {
    type: DataTypes.STRING,
    field: 'resume_content_type'
  },
  submittedAt: {
    type: DataTypes.DATE,
    field: 'submitted_at',
    defaultValue: DataTypes.NOW
  }
}, {
  tableName: 'job_applications',
  timestamps: false
});

export default Application;
export { Application };
